src/main/java/org/testeditor/web/xtext/index/resources/GlobalScopeResource.xtend

```diff
@@ -0,0 +1,186 @@
+package org.testeditor.web.xtext.index.resources
+
+import java.io.IOException
+import javax.ws.rs.Consumes
+import javax.ws.rs.POST
+import javax.ws.rs.Path
+import javax.ws.rs.Produces
+import javax.ws.rs.QueryParam
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+import org.eclipse.emf.common.util.URI
+import org.eclipse.emf.ecore.EPackage
+import org.eclipse.emf.ecore.EReference
+import org.eclipse.emf.ecore.resource.Resource
+import org.eclipse.emf.ecore.resource.ResourceSet
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl
+import org.eclipse.xtext.resource.impl.ResourceSetBasedResourceDescriptions
+import org.eclipse.xtext.scoping.IGlobalScopeProvider
+import org.eclipse.xtext.util.StringInputStream
+import org.slf4j.LoggerFactory
+
+@Path("/xtext/index/global-scope")
+@Produces(MediaType.APPLICATION_JSON)
+class GlobalScopeResource implements IGlobalScopeResource {
```

Comment:
I would not encode type of class in name, like "I" in Interface. It's a problem of naming the implementation. Proposal of C. Martin is never encode interface. When there is really no better name encode Implemenation (because these, we dont show the user). I would prefer a general name for Interface like GlobalScopeResource and a concrete name for the implementation.